src/oid4vp/holder.rs

```diff
@@ -0,0 +1,410 @@
+use super::credential_callback::*;
+use super::error::Error;
+use super::request_signer::RequestSignerInterface;
+use crate::common::*;
+use crate::credential::*;
+use crate::vdc_collection::VdcCollection;
+
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use jwt_vc::JwtVc;
+use oid4vp::core::authorization_request::parameters::ClientIdScheme;
+use oid4vp::core::credential_format::ClaimFormatDesignation;
+use oid4vp::core::presentation_definition::PresentationDefinition;
+use oid4vp::{
+    core::{
+        authorization_request::{
+            parameters::ResponseMode,
+            verification::{did::verify_with_resolver, RequestVerifier},
+            AuthorizationRequestObject,
+        },
+        metadata::WalletMetadata,
+        presentation_submission::{DescriptorMap, PresentationSubmission},
+        response::{parameters::VpToken, AuthorizationResponse, UnencodedAuthorizationResponse},
+    },
+    verifier::request_signer::RequestSigner,
+    wallet::Wallet as OID4VPWallet,
+};
+use serde_json::Value;
+use ssi::claims::jws::JwsSigner;
+use ssi::claims::vc;
+use ssi::claims::vc::syntax::IdOr;
+use ssi::dids::DIDURLBuf;
+use ssi::json_ld::iref::UriBuf;
+use ssi::JWK;
+use uniffi::deps::anyhow;
+
+/// A Holder is an entity that possesses one or more Verifiable Credentials.
+/// The Holder is typically the subject of the credentials, but not always.
+/// The Holder has the ability to generate Verifiable Presentations from
+/// these credentials and share them with Verifiers.
+#[derive(Debug, uniffi::Object)]
+pub struct Holder {
+    /// An atomic reference to the VDC collection.
+    pub(crate) vdc_collection: Arc<VdcCollection>,
+
+    /// Metadata about the holder.
+    pub(crate) metadata: WalletMetadata,
+
+    /// HTTP Request Client
+    pub(crate) client: oid4vp::core::util::ReqwestClient,
+
+    /// A list of trusted DIDs.
+    pub(crate) trusted_dids: Vec<String>,
+
+    /// A request signer callback interface to handle
+    /// cryptographic signing with native libraries.
+    pub(crate) request_signer: Arc<dyn RequestSignerInterface>,
```

Comment:
NOTE: In lieu of the KeyManager, I've added a `RequestSignerInterface` for handling the JWK signing.

This is intended to be implemented by the native code, but we may also wish to provide an implementation in Rust.
CC @Juliano1612 @cobward